### ISSUE TITLE
Permission Denied Exception when downloading Browser

### ISSUE THREAD

**Issue by CaptainDerank** [Tobi823/ffupdater#106]
Hi i tried to download Firefox Nightly and Focus through FFUpdater. When the download completes I get the attached Exception. When I try to manually install the APK from the folder, Android says that the APK is corrupted.

I am using a Samsung Galaxy Fold Z3 with Android 12.

Stacktrace:
```
java.io.FileNotFoundException: /storage/emulated/0/Android/data/de.marmaro.krt.ffupdater/files/Download/org.mozilla.fenix.apk: open failed: EACCES (Permission denied)
	at libcore.io.IoBridge.open(IoBridge.java:575)
	at java.io.FileInputStream.<init>(FileInputStream.java:160)
	at de.marmaro.krt.ffupdater.security.FileHashCalculator.createInputStream(FileHashCalculator.kt:37)
	at de.marmaro.krt.ffupdater.security.FileHashCalculator.access$createInputStream(FileHashCalculator.kt:11)
	at de.marmaro.krt.ffupdater.security.FileHashCalculator$getSHA256ofFile$2.invokeSuspend(FileHashCalculator.kt:19)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
Caused by: android.system.ErrnoException: open failed: EACCES (Permission denied)
	at libcore.io.Linux.open(Native Method)
	at libcore.io.ForwardingOs.open(ForwardingOs.java:567)
	at libcore.io.BlockGuardOs.open(BlockGuardOs.java:273)
	at libcore.io.ForwardingOs.open(ForwardingOs.java:567)
	at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:8501)
	at libcore.io.IoBridge.open(IoBridge.java:561)
	... 10 more
```
Device information:
| key | value |
| --- | --- |
| version name | 75.0.0 |
| version code | 91 |
| build type | release |
| SDK version | 31 |
| android release | 12 |
| board | lahaina |
| brand | samsung |
| device | q2q |
| manufacturer | samsung |
| model | SM-F926B |
| product | q2qxeea |
| SOC manufacturer | QTI |
| SOC model | SM8350 |
| supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| tags | release-keys |
| build time | 1640598655000 |

**Comment by Tobi823** [comment]
I am very busy right now because of the university. I'll look at the problem in the middle of next month.

**Comment by Tobi823** [comment]
Should be fixed in version 75.1.0
